python/ray/tune/logger/aim.py

```diff
@@ -0,0 +1,209 @@
+import logging
+import numpy as np
+
+from typing import TYPE_CHECKING, Dict, Optional
+
+from aim.ext.resource import DEFAULT_SYSTEM_TRACKING_INT
+from ray.tune.logger.logger import LoggerCallback
+from ray.util.debug import log_once
+from ray.tune.result import (
+    TRAINING_ITERATION,
+    TIME_TOTAL_S,
+    TIMESTEPS_TOTAL,
+)
+from ray.tune.utils import flatten_dict
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.tune.experiment.trial import Trial  # noqa: F401
+
+logger = logging.getLogger(__name__)
+
+VALID_SUMMARY_TYPES = [int, float, np.float32, np.float64, np.int32, np.int64]
+
+
+@PublicAPI
+class AimCallback(LoggerCallback):
+    """Aim Logger.
+    Logs metrics in aim format.
+    """
+
+    VALID_HPARAMS = (str, bool, int, float, list, type(None))
+    VALID_NP_HPARAMS = (np.bool8, np.float32, np.float64, np.int32, np.int64)
+
+    def __init__(
+        self,
+        repo: Optional[str] = None,
+        experiment: Optional[str] = None,
+        system_tracking_interval: Optional[int] = DEFAULT_SYSTEM_TRACKING_INT,
+        log_system_params: bool = True,
+        metrics: Optional[str] = None,
+    ):
+
+        self._repo_path = repo
+        self._experiment_name = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._metrics = metrics
+        self._log_value_warned = False
+
+        try:
+            from aim.sdk import Run
+
+            self._run_cls = Run
+            """
+            # Todo: implement the capability of images audio etc.
+            from aim import Image
+            from aim import Distribution
+            from aim.ext.resource.configs import DEFAULT_SYSTEM_TRACKING_INT
+            """
+        except ImportError:
+            if log_once("aim-install"):
+                logger.info("Please run `pip install aim` to use the AimLogger.")
+            raise
+
+        self._trial_run: Dict["Trial", Run] = {}
+
+    def _create_run(self, trial):
+        """
+        Returns: Run
+        """
+        run = self._run_cls(
+            repo=self._repo_path,
+            experiment=self._experiment_name,
+            system_tracking_interval=self._system_tracking_interval,
+            log_system_params=self._log_system_params,
+        )
+        run["trial_id"] = trial.trial_id
+        return run
+
+    def log_trial_start(self, trial: "Trial"):
+        logger.info(f"trial {trial} logger is started")
+
+        # do nothing when the trial is alrady initiliated
+        if trial in self._trial_run:
```

Comment:
I'm not sure I understand intended logic here.
The current Run (if available) is getting replaced by the new one, correct? If so, shouldn't the comment `# do nothing when the trial is alrady initiliated` change? 🤔

Reply:
@alberttorosyan , you're totally right thanks! I guess the comment can just be removed completly.